src/components/MessageInput/hooks/useCooldownTimer.tsx

```diff
@@ -36,17 +35,19 @@ export const useCooldownTimer = <
   ) as Date;
 
   useEffect(() => {
-    if (skipCooldown || !cooldownInterval || !ownLatestMessageDate) return;
-
-    const remainingCooldown = Math.round(
-      cooldownInterval - (new Date().getTime() - ownLatestMessageDate.getTime()) / 1000,
+    const timeSinceOwnLastMessage = ownLatestMessageDate
+      ? (new Date().getTime() - ownLatestMessageDate.getTime()) / 1000
+      : undefined;
+
+    setCooldownRemaining(
+      !skipCooldown && timeSinceOwnLastMessage && cooldownInterval > timeSinceOwnLastMessage
+        ? Math.round(cooldownInterval - timeSinceOwnLastMessage)
+        : 0,
```

Comment:
Why this change? It looks like it's doing the same thing just a bit harder to read on a first glance.

Reply:
You need to set also 0 cooldown remaining, what was not happening before. Now the condition states the 3 cases, when remaining cooldown is non-zero. Does not seem to me harder to read.

Reply:
What does 0 seconds cooldown remaining look like in the UI? I'd imagine 0 seconds remaining would also mean skip - or?

Reply:
yes, it means no cooldown